cpp/src/generators/erdos_renyi_generator.cu

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph/graph_generators.hpp>
+#include <cugraph/utilities/error.hpp>
+
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/copy.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/random.h>
+
+namespace cugraph {
+
+template <typename vertex_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>>
+generate_erdos_renyi_graph_edgelist(raft::handle_t const& handle,
+                                    vertex_t num_vertices,
+                                    float p,
+                                    vertex_t base_vertex_id,
+                                    uint64_t seed)
+{
+  auto random_iterator = thrust::make_transform_iterator(
+    thrust::make_counting_iterator<size_t>(0), [seed] __device__(size_t index) {
+      thrust::default_random_engine rng(seed);
+      thrust::uniform_real_distribution<float> dist(0.0, 1.0);
+      rng.discard(index);
+      return dist(rng);
+    });
+
+  size_t count = thrust::count_if(rmm::exec_policy(handle.get_stream()),
+                                  random_iterator,
+                                  random_iterator + num_vertices * num_vertices,
```

Comment:
Just want to note that the Gnp model can be way more expensive than the Gnm model if ```num_vertices``` is large (but much simpler to implement as we don't need to worry about duplicates).

Shouldn't we add gnp in the function name so users can expect this will go over ```num_vertices``` * ```num_vertices``` potential edges?

Reply:
Changed function name.  Add a gnm function as well, although not implemented in this PR.